core/src/main/scala/cats/data/NonEmptySet.scala

```diff
@@ -0,0 +1,411 @@
+    * scala> import cats.implicits._
+    * scala> val nes = NonEmptySet.of(1, 2, 4, 5)
+    * scala> nes & NonEmptySet.of(1, 2, 7)
+    * res0: scala.collection.immutable.SortedSet[Int] = TreeSet(1, 2)
+    * }}}
+    */
+  def &(as: NonEmptySet[A]): SortedSet[A] = intersect(as)
+
+
+  /**
+    * Removes a key from this set, returning a new `SortedSet`.
+    */
+  def -(a: A): SortedSet[A] = toSortedSet - a
+
+  /**
+    * Applies f to all the elements
+    */
+  def map[B: Order](f: A ⇒ B): NonEmptySet[B] =
+    NonEmptySetImpl.create(SortedSet(toSortedSet.map(f).to: _*)(Order[B].toOrdering))
+
+  /**
+    * Converts this set to a `NonEmptyList`.
+    * {{{
+    * scala> import cats.data.NonEmptySet
+    * scala> import cats.implicits._
+    * scala> val nes = NonEmptySet.of(1, 2, 3, 4, 5)
+    * scala> nes.toNonEmptyList
+    * res0: cats.data.NonEmptyList[Int] = NonEmptyList(1, 2, 3, 4, 5)
+    * }}}
+    */
+  def toNonEmptyList: NonEmptyList[A] = NonEmptyList.fromListUnsafe(toSortedSet.toList)
+
+  /**
+    * Returns the first element of this set.
+    */
+  def head: A = toSortedSet.head
+
+  /**
+    * Returns all but the first element of this set.
+    */
+  def tail: SortedSet[A] = toSortedSet.tail
+
+  /**
+    * Returns the last element of this set.
+    */
+  def last: A = toSortedSet.last
+
+  /**
+    * Alias for [[contains]]
+    * {{{
+    * scala> import cats.data.NonEmptySet
+    * scala> import cats.implicits._
+    * scala> val nes = NonEmptySet.of(1, 2, 3, 4, 5)
+    * scala> nes(3)
+    * res0: Boolean = true
+    * scala> nes(7)
+    * res1: Boolean = false
+    * }}}
+    */
+  def apply(a: A): Boolean = contains(a)
+
+  /**
+    * Tests if some element is contained in this set.
+    */
+  def contains(a: A): Boolean = toSortedSet(a)
+
+  /**
+    * Computes the difference of this set and another set.
+    */
+  def diff(as: NonEmptySet[A]): SortedSet[A] = toSortedSet -- as.toSortedSet
+
+  /**
+    * Computes the union between this NES and another NES.
+    */
+  def union(as: NonEmptySet[A]): NonEmptySet[A] = NonEmptySetImpl.create(toSortedSet ++ as.toSortedSet)
+
+  /**
+    * Computes the intersection between this set and another set.
+    */
+  def intersect(as: NonEmptySet[A]): SortedSet[A] = toSortedSet.filter(as.apply)
+
+  /**
+    * Tests whether a predicate holds for all elements of this set.
+    */
+  def forall(p: A ⇒ Boolean): Boolean = toSortedSet.forall(p)
+
+  /**
+    * Tests whether a predicate holds for at least one element of this set.
+    */
+  def exists(f: A ⇒ Boolean): Boolean = toSortedSet.exists(f)
+
+  /**
+    * Returns the first value that matches the given predicate.
+    */
+  def find(f: A ⇒ Boolean): Option[A] = toSortedSet.find(f)
+
+  /**
+    * Returns a new `SortedSet` containing all elements where the result of `pf` is defined.
+    */
+  def collect[B: Order](pf: PartialFunction[A, B]): SortedSet[B] = {
+    implicit val ordering = Order[B].toOrdering
```

Comment:
Same comment as https://github.com/typelevel/cats/pull/2141/files#r173359573

Reply:
I think there are some other places that this applies to this PR. I don't necessarily consider this a blocker, but I'd like us to at least decide what we want to do about this before we publish a release that has the name of the implicit parameter fixed (I think that the auto-generated one will be something like `ev1`).